src/database/database-mariadb.cpp

```diff
@@ -0,0 +1,1407 @@
+		throw DatabaseException("MariaDB Error");
+	}
+}
+
+
+/*
+ *
+ * PlayerDatabaseMariaDB -- The MariaDB Player Database Backend
+ * 
+ */
+
+
+PlayerDatabaseMariaDB::PlayerDatabaseMariaDB(const std::string &connect_string)
+: Database_MariaDB(connect_string, "_player"), PlayerDatabase() {
+	
+	// connect to database
+	connect();
+
+	// create tables if they do n
+	createTable("player",
+		"name VARCHAR(60),"
+		"pitch DECIMAL(15, 7) NOT NULL,"
+		"yaw DECIMAL(15, 7) NOT NULL,"
+		"posX DECIMAL(15, 7) NOT NULL,"
+		"posY DECIMAL(15, 7) NOT NULL,"
+		"posZ DECIMAL(15, 7) NOT NULL,"
+		"hp INT NOT NULL,"
+		"breath INT NOT NULL,"
+		"creation_date DATETIME NOT NULL DEFAULT NOW(),"
+		"modification_date DATETIME NOT NULL DEFAULT NOW(),"
+		"PRIMARY KEY (name)"
+	);
+
+	createTable("player_inventories",
+		"player VARCHAR(60),"
+		"inv_id INT NOT NULL,"
+		"inv_width INT NOT NULL,"
+		"inv_name TEXT NOT NULL DEFAULT '',"
+		"inv_size INT NOT NULL,"
+		"PRIMARY KEY(player, inv_id),"
+		"CONSTRAINT player_inventories_fkey FOREIGN KEY (player) REFERENCES player (name) ON DELETE CASCADE"
+	);
+
+	createTable("player_inventory_items",
+		"player VARCHAR(60),"
+		"inv_id INT NOT NULL,"
+		"slot_id INT NOT NULL,"
+		"item TEXT NOT NULL DEFAULT '',"
+		"PRIMARY KEY(player, inv_id, slot_id),"
+		"CONSTRAINT player_inventory_items_fkey FOREIGN KEY (player) REFERENCES player (name) ON DELETE CASCADE"
+	);
+
+	createTable("player_metadata",
+		"player VARCHAR(60),"
+		"attr VARCHAR(256) NOT NULL,"
+		"value TEXT,"
+		"PRIMARY KEY(player, attr),"
+		"CONSTRAINT player_metadata_fkey FOREIGN KEY (player) REFERENCES player (name) ON DELETE CASCADE"
+	);
+
+	try {
+
+		// prepare statements
+		stmtAddPlayerInventory.reset(prepareStatement(
+			"INSERT INTO player_inventories (player, inv_id, inv_width, inv_name, inv_size) "
+			"VALUES (?, ?, ?, ?, ?)"
+		));
+
+		stmtAddPlayerInventoryItem.reset(prepareStatement(
+			"INSERT INTO player_inventory_items (player, inv_id, slot_id, item) VALUES "
+			"(?, ?, ?, ?)"
+		));
+
+		stmtCreatePlayer.reset(prepareStatement(
+			"INSERT INTO player(name, pitch, yaw, posX, posY, posZ, hp, breath) "
+			"VALUES (?, ?, ?, ?, ?, ?, ?, ?)"
+		));
+
+		stmtLoadPlayer.reset(prepareStatement(
+			"SELECT pitch, yaw, posX, posY, posZ, hp, breath FROM player WHERE name = ?"
+		));
+
+		stmtLoadPlayerInventories.reset(prepareStatement(
+			"SELECT inv_id, inv_width, inv_name, inv_size "
+			"FROM player_inventories "
+			"WHERE player = ? ORDER BY inv_id"
+		));
+
+		stmtLoadPlayerInventoryItems.reset(prepareStatement(
+			"SELECT slot_id, item "
+			"FROM player_inventory_items "
+			"WHERE player = ? AND inv_id = ?"
+		));
+
+		stmtLoadPlayerList.reset(prepareStatement(
+			"SELECT name FROM player"
+		));
+
+		stmtLoadPlayerMetadata.reset(prepareStatement(
+			"SELECT attr, value FROM player_metadata WHERE player = ?"
+		));
+
+		stmtPlayerExists.reset(prepareStatement(
+			"SELECT EXISTS (SELECT * FROM player WHERE name = ?)"
```

Comment:
just SELECT name, not all fields, this will optimize fetch time DB side

Reply:
I doubt it. Keep in mind this is an `EXISTS` statement. The optimizer should definitely be smart enough to optimize this out. I see no reason to add verbosity here.

Reply:
Fixed in latest commit.